### PR TITLE
[Material] Add Titanium alloy to materials

### PR DESCRIPTION
**Pull request by vkhodygo**
Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [ x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636">pipeline status</a> for feature branch [PR_8636](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636). Pipeline [788468202](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/788468202) was triggered at [5f338fa](https://github.com/FreeCAD/FreeCAD/pull/8636/commits/5f338fa8b4bdbccdc9923ce0f58d6f92963328d3). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636">pipeline status</a> for feature branch [PR_8636](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636). Pipeline [788470974](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/788470974) was triggered at [cf2853b](https://github.com/FreeCAD/FreeCAD/pull/8636/commits/cf2853b72ef650fe27e36c83c5773510a5314eef). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
Many thanks, very valuable additon!

However, before we can put this in, did you check the license of MatWeb?: https://asm.matweb.com/search/terms.asp
I mean you set this as source so we need to assure it is compatible with the GPL license we use for FreeCAD.
I am no lawyer. I cannot read out if it is allowed or not.

But what about taking the values from different sources? This is what I do: search around the web, taking data sheets of different suppliers. I take then the mean value from the different sources. This way I don't violate any copyright and it is also my  work (the research and finding the mean values).

Besides this, it would be nice if you can also add the electrical properties like permittivity, permeability and conductivity since especially for metals this is of interest.

**Comment by berndhahnebach** [comment]
https://www.matweb.com/reference/terms.aspx

**Comment by vkhodygo** [comment]
@donovaly @berndhahnebach 

Thanks folks, I didn't expect it to be that bad. I'll find other sources and adjust the PR.

**Comment by vkhodygo** [comment]
@donovaly @berndhahnebach 
What do you think about these data sheets:

https://www.smithmetal.com/ti-6al-4v-grade-5.htm
https://www.finetubes.co.uk/products/materials/titanium-tubes/alloy-ti6al4v-uns-56400-wnr-37165
https://kyocera-sgstool.co.uk/titanium-resources/titanium-information-everything-you-need-to-know/ti-6al-4v-grade-5-titanium-alloy-data-sheet/

I couldn't find any information regarding redistribution terms though. Surprisingly, Wikipedia refers to MatWeb as well.

**Comment by vkhodygo** [comment]
https://www.tandfonline.com/doi/abs/10.1080/10940349908945686  is probably the best choice to cite, however, I'm not sure it contains all the parameters.

**Comment by donovaly** [comment]
> @donovaly @berndhahnebach What do you think about these data sheets:
> 
> https://www.smithmetal.com/ti-6al-4v-grade-5.htm https://www.finetubes.co.uk/products/materials/titanium-tubes/alloy-ti6al4v-uns-56400-wnr-37165 https://kyocera-sgstool.co.uk/titanium-resources/titanium-information-everything-you-need-to-know/ti-6al-4v-grade-5-titanium-alloy-data-sheet/

As I wrote, take all datasheets you can get, calculate the mean values for each material parameter and use this for the material card in FreeCAD. This is legally safe and you don't need to state the sources, because the source is you and your research and calculation work.

**Comment by vkhodygo** [comment]
@donovaly Done.
I [used](https://github.com/FreeCAD/FreeCAD/files/10831885/ti_props.ods) median values instead of mean, that seems to be more reasonable here.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636">pipeline status</a> for feature branch [PR_8636](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636). Pipeline [788928440](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/788928440) was triggered at [e9bb05a](https://github.com/FreeCAD/FreeCAD/pull/8636/commits/e9bb05aed3aedc9a39d67135e5cd91d712d5736b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat [hunk 1]
```diff
@@ -8,17 +8,21 @@
...
-YieldStrength = 880 MPa
-YoungsModulus = 113.8 GPa
+UltimateTensileStrength = 975.5 MPa
+YieldStrength = 910 MPa
+YoungsModulus = 114 GPa
+Hardness = 35 HRC
 
 [Thermal]
-SpecificHeat = 5266.3 J/kg/K
+SpecificHeat = 553.0 J/kg/K
```
Comment (donovaly) [inline src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat:22]: Only a tenth now, is this correct?

File: src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat [hunk 2]
```diff
@@ -8,17 +8,21 @@
...
 ThermalConductivity = 6.7 W/m/K
 ThermalExpansionCoefficient = 8.6 µm/m/K
+
+[Electromagnetic]
+ElectricalConductivity = 580012.8 S/m
+RelativePermeability = 1.00005
```
Comment (donovaly) [inline src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat:28]: At the very end of the material card, our parser needs a linebreak, can you please add it?

**Comment by donovaly** [comment]
Can you please also add the material number "3.7165" to the material card? (see the Aluminum alloy cards that specify the material number as well. This number makes it easy for users to check if they use the right material because the number is ISO standardized and thus unique.

File: src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat (hunk 2 above)
Comment (vkhodygo) [reply src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat:28]: Yes, I did the conversion incorrectly originally. All other values are in this range, should be fine now.

File: src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat (hunk 1 above)
Comment (vkhodygo) [reply src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat:22]: Yes, I did the conversion incorrectly originally. All other values are in this range, should be fine now.

File: src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat (hunk 2 above)
Comment (vkhodygo) [reply src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat:28]: Sure.

Comment (vkhodygo) [reply src/Mod/Material/StandardMaterial/Ti-6Al-4V.FCMat:28]: Sure, I should've done this earlier.

**Comment by vkhodygo** [comment]
@donovaly Fixed.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636">pipeline status</a> for feature branch [PR_8636](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8636). Pipeline [788943018](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/788943018) was triggered at [c0a9a90](https://github.com/FreeCAD/FreeCAD/pull/8636/commits/c0a9a90eb0bdc72ed76c1ce5d1a576ff706d7a70). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
Thanks.

**Comment by donovaly** [comment]
@vkhodygo , for the future: You also need to register new files to CMake, otherwise the material card will not be copied over to the build folder of those who compile FreeCAD. it would then also not be part of the release.
I did this now: #8660

**Comment by vkhodygo** [comment]
@donovaly I wasn't aware of that, many thanks for your help.

**Pull request merged**